### PR TITLE
Added endpoint to the API for universal spin table

### LINKED ISSUE #243: Universal Spin Table API Endpoint
Setup an API endpoint for a Universal Spin Table API

### PR DESCRIPTION
**Pull request by tech3371**
# Change Summary

Create endpoint for spin table query. Updated how we add route to the API and updated places that uses it.

## New Files
- sds_data_manager/lambda_code/SDSCode/spin_table_api.py
   - Place holder for API handler code

## Updated Files
- sds_data_manager/stacks/api_gateway_stack.py
   - added stack to create Lambda that handler API calls.
   - Refactor code to be able to add route easily.
- sds_data_manager/stacks/database_stack.py and sds_data_manager/stacks/domain_stack.py and sds_data_manager/stacks/ecr_stack.py
   - Added comments for future use
- sds_data_manager/stacks/sds_data_manager_stack.py
   - Updated to use new way to add route
- sds_data_manager/utils/stackbuilder.py
   - Rearrange stack to create resources it needs
   - Added new stack to create spin table endpoint

closes https://github.com/IMAP-Science-Operations-Center/imap_processing/issues/243

### CODE REVIEW

File: sds_data_manager/stacks/api_gateway_stack.py
Comment (greglucas) [inline sds_data_manager/stacks/api_gateway_stack.py]:
```suggestion
        lambda_function : lambda_.Function
            Lambda function to trigger when this route is hit.
```

File: sds_data_manager/utils/stackbuilder.py
Comment (greglucas) [inline sds_data_manager/utils/stackbuilder.py:114]:
```suggestion
    spin_api_stack = api_gateway_stack.APILambda(
```

All the other routes live in the sds_data_manager stack. Should this be moved up there too? It kind of sticks out as the only Lambda at this level right now. But maybe we can reorganize again in a follow-up PR...

**Review by greglucas — COMMENTED** [review]
Looks like the tests also need to be updated, but overall looks like a nice abstraction to me!

File: sds_data_manager/utils/stackbuilder.py
Comment (tech3371) [reply sds_data_manager/utils/stackbuilder.py:114]: I think future PR because I would like to group API lambda out. But I wanted to wait until other PR is merged.

**Review by greglucas — DISMISSED** [review]

**Review by greglucas — APPROVED** [review]

File: sds_data_manager/lambda_code/SDSCode/spin_table_api.py
Comment (bourque) [inline sds_data_manager/lambda_code/SDSCode/spin_table_api.py:5]: I reckon that you have this here for debugging purposes? Could it be removed?

File: sds_data_manager/lambda_code/SDSCode/spin_table_api.py
Comment (bourque) [inline sds_data_manager/lambda_code/SDSCode/spin_table_api.py:1]: I've been bugging everyone to add module docstrings so I will do the same here 😅

File: tests/infrastructure/test_apigateway.py
Comment (bourque) [inline tests/infrastructure/test_apigateway.py:37]: Perhaps we could define the top-level directory in the `__init__.py` file and import/use it from there, like we do in the `imap_processing` repo: https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/59b42d30453b769455e943ca9913a2cd1317f27a/imap_processing/__init__.py#L10

**Review by bourque — APPROVED** [review]
Nice work! I just have a few non-blocking comments/suggestions

File: tests/infrastructure/test_apigateway.py
Comment (tech3371) [reply tests/infrastructure/test_apigateway.py:37]: I think that's a great idea. Do you mind if I do this and function doc string in the upcoming PR?

**Comment by sdhoyt** [comment]
closes #243

**Pull request merged**
